packages/core/logging/core-logging-common-internal/src/layouts/conversions/message.ts

```diff
@@ -6,20 +6,22 @@
  * Side Public License, v 1.
  */
 
-import ansiRegex from 'ansi-regex';
 import { LogRecord } from '@kbn/logging';
 import { Conversion } from './types';
 
-// Defining it globally because it's more performant than creating for each log entry
-// We can reuse the same global RegExp here because `.replace()` automatically resets the `.lastIndex` of the RegExp.
-const ANSI_ESCAPE_CODES_REGEXP = ansiRegex();
+// From https://www.ascii-code.com/characters/control-characters,
```

Comment:
I was curios what the JSON.stringify behaviour is and it seems like it follows this spec https://github.com/tc39/proposal-well-formed-stringify here's an example implementation https://github.com/zloirock/core-js/blob/master/packages/core-js/modules/es.json.stringify.js#L22

Reply:
yeah... the problem with JSON.stringify is it escapes carriage returns as well.
But it's interesting the way they escape the codes: 
https://github.com/zloirock/core-js/blob/effda0ce8fa8e4d75d0420d7b38af48beda5c29e/packages/core-js/modules/es.json.stringify.js#L54-L60

I wonder which one we'd prefer for performance reasons.

Reply:
We would need to test to confirm it of course, but my gut feeling is that the escape introduced in this PR may be more performant than the initial approach with `ansiRegex` anyway (given the overall complexity of the ansi regexp). So I feel like we're fine merging with the PR's current approach?

But if we want to test an alternative for performances, I'm fine with it.

Reply:
I wanted to timebox a performance evaluation just in case (in the past we've found [that native APIs are sometimes slower than custom implementations](https://github.com/elastic/kibana/pull/168812#discussion_r1360521393) 🤷).

And the custom implementation is 2x faster (of course 🙄):
```
encodeURI (%1B) x 853,224 ops/sec ±0.63% (95 runs sampled)
escape charCodeAt (\u1b) x 1,956,098 ops/sec ±0.28% (98 runs sampled)
escape charCodeAt with leading zeros (\u001b) x 1,612,032 ops/sec ±0.26% (98 runs sampled)
JSON.stringify (\u001b) x 1,419,471 ops/sec ±0.23% (98 runs sampled)
Fastest is escape charCodeAt (\u1b)
```

Even `JSON.stringify` was much faster than `encodeURI` 🤯 

My follow-up question is: are we willing to pay the 17.6% penalty for injecting leading zeros (as it matches what's logged in the meta, so it may be easier to search when analyzing logs), or should we aim for performance at the cost of harder post-analysis?

Reply:
I don't think performance is that important here so we might as well maintain consistency

Reply:
In that case, I'll use the `JSON.stringify` alternative: [`03237e5` (#169773)](https://github.com/elastic/kibana/pull/169773/commits/03237e5bbe8a7cf547cc98de323d4827af39625c)